### PR TITLE
Jest unit-testing using CCF polyfill not working #109

### PR DESCRIPTION
**Pull request by Aymalla**
Resolve the issue of `jest` unable to run unit-tests when using CCF polyfill  #109

### CODE REVIEW

File: data-reconciliation-app/test/unit-test/services/ingest-service.test.ts [hunk 1]
```diff
@@ -13,30 +12,42 @@ describe("Data Ingestion Service", () => {
...
+    // Act
+    const testDataRecords: DataRecord[] = [];
+
+    // Assert
+    const result = ingestService.submitData(userId, testDataRecords);
+
+    // Assert
+    expect(result).not.toBeNull();
+    expect(result.success).toBe(true);
+    expect(result.content).toBe("data has ingested successfully");
```
Comment (julioalex-rezende) [inline data-reconciliation-app/test/unit-test/services/ingest-service.test.ts:39]: Should empty data be allowed? IMHO we could block this and treat similarly to `null` data

File: data-reconciliation-app/test/unit-test/repositories/kv-repository.test.ts [hunk 2]
```diff
@@ -1,64 +1,74 @@
-// TODO: need to use ccf polyfill to solve the issue of key-value map with unit-test
+// Use the CCF polyfill to mock-up all key-value map functionality for unit-test
 import "@microsoft/ccf-app/polyfill.js";
 
 import { randomUUID } from "crypto";
-import { IRepository } from "../../../src/repositories/kv-repository";
+import keyValueRepository, {
```
Comment (macromania) [inline data-reconciliation-app/test/unit-test/repositories/kv-repository.test.ts:5]: I think your prettier setting is still 80 characters.

**Comment by macromania** [comment]
I and @Aymalla discussed the possible other alternatives and decided to instead go with [Jest Spy](https://jestjs.io/docs/jest-object#jestspyonobject-methodname).

This will help us simply the unit tests, avoiding experimental esm/node setup.

The repository itself will not be unit tested as integration testing using `make test` already deploys the K-V Store and covers with e2e tests.

File: data-reconciliation-app/test/unit-test/repositories/kv-repository.test.ts (hunk 2 above)
Comment (Aymalla) [reply data-reconciliation-app/test/unit-test/repositories/kv-repository.test.ts:5]: I've add it as a separate story

**Review by julioalex-rezende — APPROVED** [review]
Good work. :)
Approving from my end, but not in a deep technical level related to project dependencies/config.

**Pull request merged**